libs/pageserver_api/src/models.rs

```diff
@@ -796,5 +854,30 @@ mod tests {
             "expect unknown field `unknown_field` error, got: {}",
             err
         );
+
+        let config = HashMap::from_iter(std::iter::once(("unknown_field", "unknown_value")));
+        let err = TenantConfig::deserialize_from_settings(config).unwrap_err();
+        assert!(
+            err.to_string().contains("unknown field `unknown_field`"),
+            "expect unknown field `unknown_field` error, got: {}",
+            err
+        );
+
+        let config = HashMap::from_iter(std::iter::once(("checkpoint_distance", "not_a_number")));
+        let err = TenantConfig::deserialize_from_settings(config).unwrap_err();
+        assert_eq!(err.to_string(), "invalid digit found in string");
```

Comment:
One reason why I haven't yet gotten to fixing this is, that before these were causing hard to reason errors and that's why the current form was kept.

It seems that this test case proves that the problem is still with us? Or am I not reading it correctly, was this the error before this PR?

Reply:
Before this PR, the error message can point us to the exact fields that are wrong. This PR makes the error message harder to read. Combined with the complexity of auto converting string to number / bool, I decide to put this PR on hold for now... 😂